src/parser/jobs/brd/PitchPerfectUsage.js

```diff
@@ -0,0 +1,314 @@
+			abilityId: ACTIONS.PITCH_PERFECT.id,
+		}, this._onPPDamage)
+
+		this.addHook('damage', {
+			by: 'player',
+			abilityId: [STATUSES.CAUSTIC_BITE.id, STATUSES.STORMBITE.id],
+			tick: true,
+		}, this._onDotTick)
+
+		this.addHook('cast', {
+			by: 'player',
+			abilityId: ACTIONS.THE_WANDERERS_MINUET.id,
+		}, this._onWMCast)
+
+		this.addHook('complete', this._onComplete)
+
+	}
+
+	_onDotTick(event) {
+		// Keeping track of the first dot tick of each enemy
+		// We only need to know when the first tick happens, since the frequency is known
+		const enemy = this._getEnemy(event.targetID)
+
+		enemy.tick[event.ability.guid] = event
+
+	}
+
+	_onWMCast(event) {
+		this._lastWMCast = event
+	}
+
+	_onPPDamage(event) {
+		const enemy = this._getEnemy(event.targetID)
+		const wm = this._lastWMCast
+
+		if (event.stacks === undefined) {
+			//Fuck, abort!
+			return
+		}
+
+		const ppEvent = {
+			damageEvent: event,
+			issue: NONE,
+			timeLeftOnSong: Math.max(wm.timestamp + SONG_DURATION - event.timestamp, 0),
+			critOnDot: {
+				[STATUSES.CAUSTIC_BITE.id]: 0,
+				[STATUSES.STORMBITE.id]: 0,
+			},
+			get stacks() { return this.damageEvent && this.damageEvent.stacks || undefined },
+			get timestamp() { return this.damageEvent && this.damageEvent.timestamp },
+		}
+
+		this._ppEvents.push(ppEvent)
+
+		// Only an issue if there are dot ticks left on the song and sufficient time to use PP (animation lock)
+		if (enemy.lastTick + DOT_TICK_FREQUENCY < wm.timestamp + SONG_DURATION - ANIMATION_LOCK) {
+
+			// We write down the crit on each dot, to provide the information later
+			ppEvent.critOnDot[STATUSES.CAUSTIC_BITE.id] = enemy.tick[STATUSES.CAUSTIC_BITE.id] && enemy.tick[STATUSES.CAUSTIC_BITE.id].expectedCritRate * CONVERSION_FACTOR
+			ppEvent.critOnDot[STATUSES.STORMBITE.id] = enemy.tick[STATUSES.STORMBITE.id] && enemy.tick[STATUSES.STORMBITE.id].expectedCritRate * CONVERSION_FACTOR
+
+			// If crit is above threshold for PP2
+			// Using PP1 when not at the end of the song is not ideal
+			if (
+				enemy.tick[STATUSES.CAUSTIC_BITE.id]
+				&& enemy.tick[STATUSES.STORMBITE.id]
+				&& enemy.tick[STATUSES.CAUSTIC_BITE.id].expectedCritRate * CONVERSION_FACTOR > PP2_THRESHOLD
+				&& enemy.tick[STATUSES.STORMBITE.id].expectedCritRate * CONVERSION_FACTOR > PP2_THRESHOLD
+			) {
+				// Using PP1 when not at the end of the song is not ideal
+				if (event.stacks === PP[1]) {
+					ppEvent.issue = PP1_NOT_AT_END
+				}
+			// Using PP2 when crit is below the threshold is not ideal
+			} else if (event.stacks === PP[2]) {
+
+				ppEvent.issue = PP2_ON_LOW_CRIT
+
+			// Using PP1 when not at the end of the song is not ideal
+			} else if (event.stacks === PP[1]) {
+
+				ppEvent.issue = PP1_NOT_AT_END
+
+			}
+		}
+	}
+
+	_onComplete() {
+		const badPPs = this._ppEvents.filter(pp => pp.issue !== NONE).length
+
+		if (badPPs > 0) {
+			this.suggestions.add(new TieredSuggestion({
+				icon: ACTIONS.PITCH_PERFECT.icon,
+				content: <Fragment>
```

Comment:
Nit: Can use `<>` instead of `<Fragment>` when you don't need to pass it a prop.